geopy/geocoders/arcgis.py

```diff
@@ -62,26 +63,52 @@ def __init__(self, username=None, password=None, referer=None, # pylint: disable
             through the specified proxy. E.g., {"https": "192.0.2.0"}. For
             more information, see documentation on
             :class:`urllib2.ProxyHandler`.
+            
+        :param string client_id: ArcGIS API client ID. Required to authenticate
+            with OAuth2
+            
+        :param string client_secret: ArcGIS API client secret. Required to 
+            authenticate with OAuth2
         """
         super(ArcGIS, self).__init__(
             scheme=scheme, timeout=timeout, proxies=proxies, user_agent=user_agent
         )
-        if username or password or referer:
-            if not (username and password and referer):
-                raise ConfigurationError(
-                    "Authenticated mode requires username,"
-                    " password, and referer"
-                )
+        if username or password or referer or client_id or client_secret:
```

Comment:
I think it would be cleaner to introduce these two vars:

```python
is_oauth = client_id or client_secret
is_username_auth = username or password or referer
```